src/app/parcels/add/AddParcelForm.tsx

```diff
@@ -0,0 +1,196 @@
+import CollectionCentreCard from "@/app/parcels/add/formSections/CollectionCentreCard";
+import { insertParcel } from "@/app/parcels/add/databaseFunctions";
+
+interface AddParcelFields {
+    voucherNumber: string;
+    packingDate: Date | null;
+    timeOfDay: Date | null;
+    shippingMethod: string | null;
+    collectionDate: Date | null;
+    collectionTime: Date | null;
+    collectionCentre: string | null;
+}
+
+const withCollectionFormSections = [
+    VoucherNumberCard,
+    PackingDateCard,
+    TimeOfDayCard,
+    ShippingMethodCard,
+    CollectionDateCard,
+    CollectionTimeCard,
+    CollectionCentreCard,
+];
+
+const noCollectionFormSections = [
+    VoucherNumberCard,
+    PackingDateCard,
+    TimeOfDayCard,
+    ShippingMethodCard,
+];
+
+const initialFields: AddParcelFields = {
+    voucherNumber: "",
+    packingDate: null,
+    timeOfDay: null,
+    shippingMethod: "",
+    collectionDate: null,
+    collectionTime: null,
+    collectionCentre: null,
+};
+
+const initialFormErrors: FormErrors = {
+    voucherNumber: Errors.none,
+    packingDate: Errors.initial,
+    timeOfDay: Errors.initial,
+    shippingMethod: Errors.initial,
+    collectionDate: Errors.initial,
+    collectionTime: Errors.initial,
+    collectionCentre: Errors.initial,
+};
+
+const AddParcelForm: React.FC<{ id: string }> = (props: { id: string }) => {
+    const router = useRouter();
+    const [fields, setFields] = useState<AddParcelFields>(initialFields);
+    const [formErrors, setFormErrors] = useState(initialFormErrors);
+    const [submitError, setSubmitError] = useState(Errors.none);
+    const [submitErrorMessage, setSubmitErrorMessage] = useState("");
+    const [submitDisabled, setSubmitDisabled] = useState(false);
+
+    const formSections =
+        fields.shippingMethod === "Collection"
+            ? withCollectionFormSections
+            : noCollectionFormSections;
+
+    const fieldSetter = setField(setFields, fields);
+    const errorSetter = setError(setFormErrors, formErrors);
+
+    const submitForm = async (): Promise<void> => {
+        setSubmitDisabled(true);
+        let inputError;
+        if (fields.shippingMethod === "Collection") {
+            inputError = checkErrorOnSubmit(formErrors, setFormErrors);
+        } else {
+            inputError = checkErrorOnSubmit(formErrors, setFormErrors, [
+                "voucherNumber",
+                "packingDate",
+                "timeOfDay",
+                "shippingMethod",
+            ]);
+        }
+        if (inputError) {
+            setSubmitError(Errors.submit);
+            setSubmitDisabled(false);
+            return;
+        }
+        if (fields.packingDate === null || fields.timeOfDay === null) {
+            setSubmitDisabled(false);
+            return;
+        }
+
+        const newPackingDate = new Date(fields.packingDate);
+        const newTimeofDay = new Date(fields.timeOfDay);
+
+        const packingDateTime = new Date(
```

Comment:
You are currently doing this logic twice. Is it worth it to separate into a function?

Reply:
I think you are right!